WhenPushComesToShove/Assets/3-Scripts/Utilities/Extensions.cs

```diff
@@ -11,10 +13,53 @@ public static class Extensions
     /// <param name="maxMagnitude">Maximum magnitude of returned vector</param>
     public static Vector2 Clamp(this Vector2 v, float maxMagnitude)
     {
-        if(v.SqrMagnitude() > (maxMagnitude * maxMagnitude))
+        if (v.SqrMagnitude() > (maxMagnitude * maxMagnitude))
         {
             return v.normalized * maxMagnitude;
         }
         return v;
     }
+
+    //https://answers.unity.com/questions/530178/how-to-get-a-component-from-an-object-and-add-it-t.html
+    public static T GetCopyOf<T>(this T comp, T other) where T : Component
+    {
```

Comment:
Could you add some comments here cause I have no idea what's happening on the code alone